addons/bus/websocket.py

```diff
@@ -1120,9 +1121,35 @@ def _serve_forever(cls, websocket, db, httprequest, version):
 
 def _kick_all(code=CloseCode.GOING_AWAY):
     """ Disconnect all the websocket instances. """
+    _logger.info('disconnecting all websockets (%s)...', len(_websocket_instances))
+    count = 0
+    wait_threshold = max(odoo.sql_db._Pool._maxconn // 8, 8) if odoo.sql_db._Pool else 32
     for websocket in _websocket_instances:
         if websocket.state is ConnectionState.OPEN:
-            websocket.close(code)
+            try:
+                websocket._disconnect(code)
+            except Exception:
+                # race condition: it's possible the socket was already closed
+                continue
+            if (count + 1) % wait_threshold == 0:
+                time.sleep(0.5)
+            count += 1
+
+    # enforce the 15s timeout
```

Comment:
This should be done as part of the event loop. So we should fix `get_messages` instead of implementing our own timeout enforcement in `_kick_all`. 

```py
if self._timeout_manager.has_timed_out():
    if self.state is ConnectionState.OPEN:
        self._disconnect(
            CloseCode.ABNORMAL_CLOSURE
            if self._timeout_manager.timeout_reason is TimeoutReason.NO_RESPONSE
            else CloseCode.KEEP_ALIVE_TIMEOUT
        )
    else:
        self._terminate()
    continue
```

`TimeoutManager.TIMEOUT` is already set to 15s, so as stated earlier, we just need to ensure that the event loop wakes up after that time to check if the answer was received.